modules/programs/contour.nix

```diff
@@ -0,0 +1,1193 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.programs.contour;
+
+  filteredKeybindings =
+    map (keybind: filterAttrs (_: v: v != null) keybind) cfg.keybindings;
+in {
+  meta.maintainers = [ maintainers.ivan770 ];
+
+  options.programs.contour = {
+    enable = mkEnableOption "Contour terminal";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.contour;
+      defaultText = literalExpression "pkgs.contour";
+      description = ''
+        Contour package to install
+      '';
+    };
+
+    settings = mkOption {
```

Comment:
You can replace this very long definition of `settings` with one using `formats.yaml` like in the Pueue module: https://github.com/nix-community/home-manager/blob/1a8e35d2e53ed2ccd9818fad9c9478d56c655661/modules/services/pueue.nix#L19-L33

Reply:
I'm not sure if that's a preferred way, since, well, we lose all the type info

Reply:
Unfortunately, it is the preferred way to keep the option list maintainable. Exceptions can be made when the configuration options are very stable (such as for SSH or i3) but it seems to me that this may not be the case with Contour.

If you want your can make the settings option into a freeform module with explicit options specified for a small number of keys that most people would want to edit.